lib/src/main/scala/spinal/lib/Stream.scala

```diff
@@ -906,13 +956,19 @@ class StreamCCByToggle[T <: Data](dataType: T, inputClock: ClockDomain, outputCl
   }
 }
 
-object StreamDispatcherSequencial{
+/**
+ * This is equivalent to a StreamDemux, but with a counter attached to the port selector.
+ */
+// TODO is there any reason to have a proper class for this if we simply could reuse StreamDemux instead? ~piegames
+// TODO If there is a StreamDispatcherSequential, there should be a StreamCombinerSequential as well (name open to suggestions) ~piegames
```

Comment:
StreamCombinerSequential basicaly already exists as a StreamArbiter with the StreamArbiter.sequentialOrder arbitration :)

Reply:
I think I completely lost track about which stream modifications exist, what they do and how to name them. All my attempts to order these in some way and make it consistent failed because stream operations are heavily asymmetric (e.g. most operations have an "inverse", but it is not really the *inverse* because it does a few things differently).

Reply:
Just be care full to not break compatibility with previous releases, also things do not realy need to be 100 % perfect :)